sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs

```diff
@@ -38,9 +38,9 @@ public DiagnosticsOptions()
         /// </summary>
         public int LoggingContentSizeLimit { get; set; } = 4 * 1024;
 
-        internal IList<string> LoggingAllowedHeaderNames { get; }
+        public IList<string> LoggingAllowedHeaderNames { get; }
```

Comment:
Should document these and the other public members.

Reply:
And you should make a public constants somewhere to indicate that "*" means everything.